compiler/src/typed/typed_well_formedness.re

```diff
@@ -228,9 +245,21 @@ module WellFormednessArg: TypedtreeIter.IteratorArgument = {
         )
           when func == "==" || func == "!=" =>
         if (List.exists(((_, arg)) => exp_is_wasm_unsafe(arg), args)) {
+          let typeName =
+            switch (args) {
+            | [(_, arg), _] =>
+              switch (resolve_unsafe_type(arg)) {
+              | Some(typeName) => typeName
+              // This should never be hit
+              | None => failwith("Impossible: exp_is_wasm_unsafe > typeName")
```

Comment:
You can move this into the resolve_unsafe_type function and avoid working with options

Reply:
See above